### DISCUSSION TITLE
custom extensions directory in theiaide

### DISCUSSION THREAD

**Discussion by goyalyashpal** [eclipse-theia/theia#13706] [Q&A] [answered]
similar to how the environment variable `VSCODE_EXTENSIONS` is used by both `vscodium` and `vscode` as a directory where they store & use their extension from,

so, is it possible to direct theia-ide too to use a custom directory for storing its extensions?
if it is able to use the existing extensions from vscodium (which already uses from open-vsx) then it will be awesome :)

i have read this https://theia-ide.org/docs/user_install_vscode_extensions
but didn't find anything relevant.

### environment & versions:

* theiaide: 1.49.100
* VS Code API Version: 1.88.1
* package from: https://download.eclipse.org/theia/ide/latest/windows/TheiaIDESetup-1.49.100.exe
* os: win 10 22h2

**Comment by msujew** [comment, marked as answer]
Hey @goyalyashpal,

looking into the code, it seems like the paths are pretty much hard-coded (relative to the user directory). See [here](https://github.com/eclipse-theia/theia/blob/38eb31945130bb68fc793d4291d8d9f416541cdb/packages/plugin-ext-vscode/src/common/plugin-vscode-environment.ts#L31-L37). So as of now, it doesn't seem to be possible to select the directory via an environment variable.

Any contribution to adjust this is appreciated :)
